### DISCUSSION TITLE
Security of the terminal

### DISCUSSION THREAD

**Discussion by danbulant** [GitSquared/edex-ui#1037] [Questions] [answered]
How secure is using the terminals on a websocket port? Is there a way websites could try connecting to one of the used ports and get terminal access?

**Comment by GitSquared** [comment, marked as answer]
In theory, yes, a malicious website could connect to edex's internal websocket connection via client-side javascript, and get terminal access.
In practice however, the internal websocket server only accepts one connection so the website has very limited time to try to connect before edex's own frontend does.

**Reply by danbulant** [reply]
Oh thanks, it was just something that crossed my mind when I was looking at the source.
